zebrad/src/commands/start/sync.rs

```diff
@@ -227,31 +230,37 @@ where
                 .map_err(|e| eyre!(e))?
                 .call(zn::Request::BlocksByHash(set));
 
-            let mut verifier = self.verifier.clone();
+            let verifier = self.verifier.clone();
 
             let _ = tokio::spawn(async move {
-                match async move {
+                let result_fut = async move {
+                    let mut handles = FuturesUnordered::new();
                     let resp = request.await?;
 
                     if let zn::Response::Blocks(blocks) = resp {
                         debug!(count = blocks.len(), "received blocks");
 
                         for block in blocks {
-                            let _hash = verifier.ready_and().await?.call(block).await?;
+                            let mut verifier = verifier.clone();
+                            let handle = tokio::spawn(async move {
+                                verifier.ready_and().await?.call(block).await
+                            });
+                            handles.push(handle);
                         }
                     } else {
                         debug!(?resp, "unexpected response");
```

Comment:
I think this should be an error on par with a failure of the request. If we're erroring out on request failures, we should also error out on wrong answers.  Either way we can't advance.

Reply:
Let's move this to a separate issue: https://github.com/ZcashFoundation/zebra/pull/540/files/26edd5490787c7e4ba0b2f32590baa2e9cc7539b..4cf3f6e4416c029c6ef63072bfeef64a6c4b6fa1#r447276781

Reply:
moved to https://github.com/ZcashFoundation/zebra/issues/557